File: auth-server/src/test/java/no/vicx/authserver/config/UserDetailsConfigIntegrationTest.java
package no.vicx.authserver.config;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.security.core.userdetails.UserDetailsService;

import static org.junit.jupiter.api.Assertions.assertNotSame;

@SpringBootTest
class UserDetailsConfigIntegrationTest {

    @Autowired
    UserDetailsService userDetailsService;

    @Test
    void loadUserByUsername_givenDefaultUsername_expectNewInstanceOnEveryInvocation() {
        var firstDefaultUser = userDetailsService.loadUserByUsername("user1");
        var secondDefaultUser = userDetailsService.loadUserByUsername("user1");

        assertNotSame(firstDefaultUser, secondDefaultUser);
    }
}

File: auth-server/src/test/kotlin/no/vicx/authserver/config/UserDetailsConfigIntegrationTest.kt
package no.vicx.authserver.config

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.types.shouldNotBeSameInstanceAs
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.security.core.userdetails.UserDetailsService

@SpringBootTest
class UserDetailsConfigIntegrationTest(
    userDetailsService: UserDetailsService,
) : StringSpec({

        "when loading user details a second time, it should not be the same instance as the first one" {
            val firstDefaultUser = userDetailsService.loadUserByUsername("user1")
            val secondDefaultUser = userDetailsService.loadUserByUsername("user1")

            firstDefaultUser shouldNotBeSameInstanceAs secondDefaultUser
        }
    })
